### PR TITLE
Adding support for the old leader-elector

### PR DESCRIPTION
**Pull request by tvoran**
Adds an option to deploy the old leader-elector container that was removed in PR #568. (The new vault-k8s uses an internal mechanism for leader determination, so this is just for backwards compatibility, and can be removed in the near future.)

Note that this defaults to **not** deploying the old leader-elector container and Endpoint, unless `injector.leaderElector.useContainer` is set to `true`.

### CODE REVIEW

File: templates/injector-leader-endpoint.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# This is created here so it can be cleaned up easily, since if
+# the endpoint is left around the leader won't expire for about a minute.
+apiVersion: v1
+kind: Endpoints
+metadata:
+  name: {{ template "vault.fullname" . }}-agent-injector-leader
+  labels:
+    app.kubernetes.io/name: {{ include "vault.name" . }}-agent-injector
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
```
Comment (benashz) [inline templates/injector-leader-endpoint.yaml:11]: I wonder if adding an extra annotation denoting the deprecation of this container would be a good idea. Not sure if that would satisfy what @calvn requested w.r.t. facilitating debugging.

Comment (tvoran) [reply templates/injector-leader-endpoint.yaml:11]: Sure, something like `deprecated: true`?

I also added a change to the logging in https://github.com/hashicorp/vault-k8s/pull/297 to make it easier to tell which leader elector scheme is in use.

Comment (benashz) [reply templates/injector-leader-endpoint.yaml:13]: Sure, that sounds good.

**Review by tomhjp — APPROVED** [review]
Shall we pick a version now that we'll remove it again, so we can give advanced notice in the next changelog?

**Comment by tvoran** [comment]
@tomhjp That sounds like a good idea; how about v0.18.0? https://github.com/hashicorp/vault-helm/blob/master/CHANGELOG.md

**Comment by tomhjp** [comment]
@tvoran v0.18.0 sounds good to me, gives a nice amount of overlap and warning.

**Pull request merged**